### ISSUE TITLE
Eclipse Oxygen installation problem

### ISSUE THREAD

**Issue by cbruchert** [groovy/groovy-eclipse#422]
When I try to install it using http://dist.springsource.org/release/GRECLIPSE/e4.7, Eclipse complains with 
 "The installation cannot be completed as requested" and opens Install Remediation Page window offering to Upgrade/Downgrade Mylyn WikiText.

**Comment by eric-milles** [comment]
Which Eclipse Package did you start with and what version?

**Comment by cbruchert** [comment]
I've fresh  installed Eclipse Oxygen p2 and tried to add groovy-eclipse using the URL mentioned before.

**Comment by eric-milles** [comment]
Did you use one of these? If so, which one?

![packages](https://user-images.githubusercontent.com/18193802/34997694-aa9d0cac-faa2-11e7-9c13-f5842e1edd6e.png)

**Comment by cbruchert** [comment]
Yes, it was C/C++ package.

**Comment by eric-milles** [comment]
Starting from _Eclipse IDE for C/C++ developers_, I first run _Help > Check for Updates_.  Several C/C++ components have updates available.  From there, Groovy-Eclipse requires _Eclipse Java Development Tools_.  So I get that from the _Oxygen_ update site.  Lastly, I add the _Groovy-Eclipse Feature_ from http://dist.springsource.org/release/GRECLIPSE/e4.7 and all is well.

**Comment by cbruchert** [comment]
Now it worked, it was the Java Development Tools that was missing. The original error was very misleading. Thanks Eric!

**Comment by Humancell** [comment]
I know this is closed, but wanted to note that I hit this exact same issue ... is there a way to add the Eclipse Java Development Tools as a dependency?  Or update the documentation to make this more obvious?

I had installed the C/C++ and PHP components ... but not Java.

**Comment by eric-milles** [comment]
Sorry for the troubles.  Yes, JDT is indicated as a dependency in the feature config.  Unfortunately because a component of JDT is patched, the install does not indicate the missing dependency.

When you say update the documentation, where are you looking for steps?  I'm not sure exactly where there are instructions.  I can add a note in the install dialog, but it is quite often overlooked.

**Comment by eric-milles** [comment]
For reference, you can see here where JDT feature is indicated as required:
```xml
<feature id="org.codehaus.groovy.jdt.patch"
      label="%featureName"
      version="3.0.0.qualifier"
      provider-name="%providerName">

   ...

   <requires>
      <import feature="org.eclipse.jdt" version="3.13.3.v20180215-0725" patch="true"/>
   </requires>

   <plugin
         id="org.eclipse.jdt.core"
         download-size="0"
         install-size="0"
         version="0.0.0"
         unpack="false"
   />

</feature>
```
from https://github.com/groovy/groovy-eclipse/blob/master/jdt-patch/e47/Feature-org.codehaus.groovy.jdt.patch/feature.xml

**Comment by Humancell** [comment]
Thanks ... after re-reading the Wiki I did find the following sentence inside of the "How To Install" section:

> Since Groovy-Eclipse relies on Eclipse's Java tooling (the JDT), almost any flavor of Eclipse that includes the JDT will be compatible.

Of course since it wasn't highlighted I missed that, and the error that Eclipse presented did not make that an obvious solution.  Maybe making a note that is slightly more visible will help others.  This issue did give me the pointer to what I had to do ... and it's working great now!
